services/userService.js

```diff
@@ -0,0 +1,45 @@
+require("dotenv").config();
+
+const jwt = require("jsonwebtoken");
+const axios = require("axios");
+
+const userDao = require("../models/userDao");
+
+const kakaoLogin = async (kakaoToken) => {
+  const { data } = await axios.get("https://kapi.kakao.com/v2/user/me", {
+    headers: {
+      authorization: `Bearer ${kakaoToken}`,
+      "Content-type": "application/x-www-form-urlencoded;charset=utf-8",
+    },
+  });
+
+  const nickname = data.properties.nickname;
+  const email = data.kakao_account.email;
+  const socialId = data.id;
+
+  const { userCheck } = await userDao.getUserExistsBySocialId(socialId);
+
+  if (userCheck === "0") {
+    await userDao.createSocialUser(nickname, email, socialId);
+
+    const jsonwebtoken = await jwt.sign(
+      { kakaoId: socialId },
+      process.env.JWT_SECRET
+    );
+
+    return jsonwebtoken;
+  }
+
+  const user = await userDao.getSocialUser(socialId);
+
+  const jsonwebtoken = await jwt.sign(
+    { kakaoId: user.social_id },
+    process.env.JWT_SECRET
+  );
+
+  return jsonwebtoken;
```

Comment:
피드백드립니다.

1. userCheck를 하기위해 dao method를 하나 더 만드는 것 보다 getUserBySocialId 하나만 사용하여 변수에 담고 이어서 활용하며 작성하는 것이 더 낫습니다.
2. createSocailUser라는 변수명 보다 createUser라고 지어주세요.
3. 토큰의 payload에는 property name도, 값도 카카오의 값을 담는것이 아닌 우리 서버의 id를 담아야합니다. 카카오만 있는 것이 아닌 네이버, 구글도 붙여서 지원한다고 하면 토큰을 verify할 때 구분할 수 없음은 당연하고, 서버토큰이라는것은 해당 서버의 데이터로 만들어져야합니다!

아래 예시를 남겨드릴테니 어떻게 수정될 수 있는지 비교해보시면 도움될 것 같습니다.

```suggestion
  const user = await userDao.getUserBySocialId(socialId);

  if (!user) {
    await userDao.createUser(nickname, email, socialId);

    return jwt.sign(
      { id: user.insertId },
      process.env.JWT_SECRET
    );
  }

  return jwt.sign(
    { id: user.id },
    process.env.JWT_SECRET
  );
```